sys/kern/kern_exec.c

```diff
@@ -1202,7 +1202,11 @@ exec_new_vmspace(struct image_params *imgp, struct sysentvec *sv)
 	error = vm_map_stack(map, stack_addr, (vm_size_t)ssiz, stack_prot,
 	    stack_prot, MAP_STACK_GROWS_DOWN);
 	if (error != KERN_SUCCESS) {
-		vm_map_reservation_delete(map, cheri_kern_getbase(stack_addr));
+#ifdef __CHERI_PURE_CAPABILITY__
+		vm_map_reservation_delete(map, cheri_getbase(stack_addr));
```

Comment:
This is a bit weird. Why do the semantics differ? Surely the offset of the capability should be 0 and if it's not then the hybrid version is buggy?

Reply:
This was necessary before we had a variable `p_usrstack` for the process. I will get it out of the way in a separate commit, if it is a small patch I'll push it in this PR.

Reply:
Probably makes more sense to have it in a separate reservation patch cleanup.